### PR TITLE
reviewdog test

### PR DESCRIPTION
**Pull request by ras0219-msft**

### CODE REVIEW

File: .github/workflows/blank.yml [hunk 1]
```diff
@@ -1,38 +1,86 @@
...
-          git stash -u && git stash drop
-          reviewdog -f=diff -f.diff.strip=1 -reporter=github-pr-review < "${TMPFILE}"
+          git config user.email "none@none.none"
+          git config user.name "none"
+
+          ./vcpkg format-manifest ports/*/vcpkg.json
+          git diff >.github-pr.format-manifest
+          git add -u
+          git commit -m "tmp" --allow-empty
+          ./vcpkg x-add-version --overwrite-version --all --skip-formatting-check
```
Comment (BillyONeal) [inline .github/workflows/blank.yml]: I don't think defaulting to something as destructive as --overwrite-version is appropriate -- the vast majority of PRs must *not* use overwrite-version. Failing with a message like 'you must overwrite or choose a new version' in the event the non overwrite-version version fails is OK.

Comment (BillyONeal) [reply .github/workflows/blank.yml]: Oh, I see, we never try to push this change back to the PR.

File: .github/workflows/blank.yml [hunk 2]
```diff
@@ -1,38 +1,86 @@
-# This is a basic workflow to help you get started with Actions
+name: Formatting Review
```
Comment (BillyONeal) [inline .github/workflows/blank.yml]: Probably need a comment to the effect of "we also do these checks in the normal azure pipelines run and do not consider effects of this one 'normative'"?

File: .github/workflows/blank.yml [hunk 3]
```diff
@@ -1,38 +1,86 @@
...
+          script: |
+            const { promises: fs } = require('fs')
+            const add_version = (await fs.readFile('.github-pr.x-add-version', 'utf8')).trim()
+            const format = (await fs.readFile('.github-pr.format-manifest', 'utf8')).trim()
+
+            var output = ''
+            if (format !== "") {
+              output += "Did you remember to format all modified manifest files?\n"
+              output += "`./vcpkg format-manifest ports/*/vcpkg.json`\n"
+              output += "```diff\n" + format + "\n```\n\n"
```
Comment (BillyONeal) [inline .github/workflows/blank.yml]:
```suggestion
              output += "You need to format any edited manifest files.\n"
              output += "`./vcpkg format-manifest ports/*/vcpkg.json`\n"
              output += "```diff\n" + format + "\n```\n\n"
```

File: .github/workflows/blank.yml [hunk 4]
```diff
@@ -1,38 +1,86 @@
...
+            }
+            if (add_version !== "") {
+              output += "Did you remember to update the version database after committing all other changes?\n"
+              output += "`./vcpkg x-add-version --all --overwrite-versions`\n"
+              output += "```diff\n" + add_version + "\n```\n\n"
```
Comment (BillyONeal) [inline .github/workflows/blank.yml]:
```suggestion
              output += "You need to update the version database after committing other changes.\n"
              output += "If you added a new version, use:\n"
              output += "`./vcpkg x-add-version --all`\n"
              output += "If you updated an existing version not yet merged to microsoft/vcpkg/master, use:\n"
              output += "`./vcpkg x-add-version --all --overwrite-version`\n"
              output += "`The changes should look something like:\n"
              output += "```diff\n" + add_version + "\n```\n\n"
```

**Review by BillyONeal — COMMENTED** [review]
I think the 'did you x?' form suggests that the user should have known that they needed to do X, but that isn't going to be true for most contributors

**Comment by ras0219-msft** [comment]
See #7 for the continuation of this testing

**Pull request merged**